src/controllers/messages/users/buyer.js

```diff
@@ -164,9 +165,40 @@ async function removeUserFromQueue(recipient, listingId, title) {
   }
 }
 
+function initializeQueueHandler(listingId) {
+  const queueRef = db.ref(`listings/${listingId}/queue`);
+  queueRef.on("value", async snapshot => {
+    const queue = snapshot.val();
+    console.log(queue);
+    if (queue) {
```

Comment:
Sorry, just thought about this some more -- wouldn't `queue` always be false because it's an empty array at this point? Should we be checking `has_queue` instead?

Reply:
I used the `on` handler here, which means that the callback gets called every time the value changes.